modules/function/src/main/java/com/opengamma/strata/function/calculation/deposit/TermDepositMeasureCalculations.java

```diff
@@ -114,30 +122,19 @@ private static MultiCurrencyAmount calculatePv01(
       ExpandedTermDeposit product,
       CalculationMarketData marketData) {
 
-    return ratesProviderStream(marketData)
-        .map(provider -> calculateBucketedPv01(product, provider))
-        .collect(toScenarioResult());
+    return FxConvertibleList.of(
```

Comment:
I'm not sure about explicitly calling `FxConvertibleList.of`, it seems like a step backwards in usability.

Previously you didn't have to consider whether the currencies in your result could be automatically converted. When you called `toScenarioResult` the system checked each of the results and wrapped them in an `FxConvertibleList` if they all implemented `FxConvertible`, otherwise is used a `DefaultScenarioResult`.

Maybe there should be a method `ScenarioResult.of` which does the same thing.

I did consider taking it a step further and automatically using `CurrencyValuesArray` when all the values were `CurrencyAmount` and `MultiCurrencyValuesArray` for `MultiCurrencyAmount`. To be honest I can't remember why I decided against that, but it's possible things have changed since then and we could consider it now.